gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java

```diff
@@ -220,8 +227,21 @@ void runInternal() {
           int numPartitions = 0;
           String jobTag = defaultHelixInstanceTags;
           if (jobContext != null) {
-            inUseInstances.addAll(jobContext.getPartitionSet().stream().map(jobContext::getAssignedParticipant)
-                .filter(Objects::nonNull).collect(Collectors.toSet()));
+            log.debug("JobContext {} num partitions {}", jobContext, jobContext.getPartitionSet().size());
+
+            inUseInstances.addAll(jobContext.getPartitionSet().stream().map(i -> {
+              if (jobContext.getPartitionNumAttempts(i) > THRESHOLD_NUMBER_OF_ATTEMPTS_FOR_LOGGING) {
+                log.warn("Helix task {} has been retried for {} times, please check the config to see how we can handle this task better", jobContext.getTaskIdForPartition(i), jobContext.getPartitionNumAttempts(i));
+              }
+              if (!UNUSUAL_HELIX_TASK_STATES.contains(jobContext.getPartitionState(i))) {
+                return jobContext.getAssignedParticipant(i);
+              } else {
```

Comment:
Nit: This else is redundant, IMO it makes sense to remove it especially since we are 5 indentation levels deep already. 

Nit2: This method is a super method, and I'd encourage extracting some of the logic here. But as-is it's still readable to me (maybe not to a new comer)